src/ucm/cuda/install.c

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2001-2017.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifdef HAVE_CONFIG_H
+#  include "config.h"
+#endif
+
+#include "cudamem.h"
+
+#include <ucm/api/ucm.h>
+#include <ucm/event/event.h>
+#include <ucm/util/log.h>
+#include <ucm/util/reloc.h>
+#include <ucm/util/ucm_config.h>
+#include <ucs/sys/math.h>
+
+#include <cuda.h>
+#include <cuda_runtime.h>
+#include <unistd.h>
+#include <pthread.h>
+
+
+
+static ucm_reloc_patch_t ucm_cudamem_symbol_patches[] = {
+    {"cudaFree",   ucm_override_cudaFree},
```

Comment:
why need an array? we have only one function to patch

Reply:
just to able to add new funcs in future  if needed

Reply:
then we can add the array in the future when it would be needed

Reply:
removed